Src/VsSpecific/VsSpecific/CSharpScriptExecutor.cs

```diff
@@ -0,0 +1,115 @@
+﻿#if VS_SPECIFIC_2017 || VS_SPECIFIC_2019
+
+using Microsoft.CodeAnalysis.CSharp.Scripting;
+using Microsoft.CodeAnalysis.Scripting;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using Vim.Interpreter;
+
+namespace Vim.VisualStudio.Specific
+{
+    internal sealed class CSharpScriptExecutor : ICSharpScriptExecutor
+    {
+        private const string ScriptFolder = "vsvimscripts";
+        private Dictionary<string, Script<object>> _scripts = new Dictionary<string, Script<object>>(StringComparer.OrdinalIgnoreCase);
+        private ScriptOptions _scriptOptions = null;
+
+        private void Execute(IVim vim, CallInfo callInfo, bool createEachTime)
+        {
+            try
+            {
+                Script<object> script;
+                if (!TryGetScript(vim, callInfo.Name, createEachTime, out script))
+                    return;
+
+                var globals = new CSharpScriptGlobals(callInfo, vim);
+                script.RunAsync(globals).Wait();

```

Comment:
Calling `.Wait` can introduce some interesting deadlock cases. Did you consider having the `Execute` method return a `Task` instead so the caller could potentially wait this in a safe fashion? I'm not 100% sure that's possible here but want to make sure it was considered.

Reply:
I did not recognize the possibility of deadlock.
I will verify this problem.